### PR TITLE
"draw" images into generic io.Writer

### PR DESCRIPTION
**Pull request by Trane9991**
Hey,
In this PR I'm updating the "draw" functions to write images into generic  `io.Writer` interface, instead of `http.ResponseWriter` to be able to use them not only in HTTP handlers

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,4 +22,7 @@ _testmain.go
 *.exe
 *.test
 *~
-*.DS_Store
\ No newline at end of file
+*.DS_Store
+
+*_got
```
Comment (santiaago) [inline .gitignore]: Why do we need this ignore lines now?

File: draw/spaceinvaders/spaceinvaders_test.go [hunk 2]
```diff
@@ -2427,21 +2428,22 @@ func TestSVGGeneratedBySpaceInvaders(t *testing.T) {
...
 		md5Key := fmt.Sprintf("%x", h.Sum(nil)[:])
 
-		SpaceInvaders(rec, md5Key, mapOfColors["frogideas"][:2], 220)
-		if rec.Code != http.StatusOK {
-			t.Errorf("returned %v. Expected %v.", rec.Code, http.StatusOK)
-		}
-		if rec.Body.String() != expected {
+		SpaceInvaders(w, md5Key, mapOfColors["frogideas"][:2], 220)
+
+		// t.Log(w.String())
```
Comment (santiaago) [inline draw/spaceinvaders/spaceinvaders_test.go]: Could you remove this unnecessary comments?

File: .gitignore (hunk 1 above)
Comment (Trane9991) [reply .gitignore]: when spaceinviders tests fail it generates the `<name>_got` and `<name>_expected` svg files, I got plenty of them while was fixing tests, so it made sense to add this files to the `.gitignore` to accidentally not commit them.

File: draw/spaceinvaders/spaceinvaders_test.go (hunk 2 above)
Comment (Trane9991) [reply draw/spaceinvaders/spaceinvaders_test.go]: Oops, will do in a sec.

File: .gitignore (hunk 1 above)
Comment (Trane9991) [reply .gitignore]: `*.DS_Store` - I think VS Code automatically added it.

**Comment by santiaago** [comment]
thanks for your PR @Trane9991

**Comment by santiaago** [comment]
how are you thinking of using this now @Trane9991 ?

I always wanted to create a repo `tinygraphs.com` only for the frontend code and the controllers and make this repo only be for generating the images (but never found the time to do that). That way when you go get `github.com/taironas/tinygraphs` you only get the libraries to generate the images not the full application.

**Comment by santiaago** [comment]
I see - ok cool 👍

**Pull request merged**